src/SharedUtilities/boundary_conditions.jl

```diff
@@ -0,0 +1,75 @@
+export boundary_flux,
+    AbstractBC,
+    AbstractBoundary,
+    TopBoundary,
+    BottomBoundary,
+    diffusive_flux,
+    get_Δz
+
+"""
+    AbstractBC{FT <: AbstractFloat}
+An abstract type for types of boundary conditions, which will include
+prescribed functions of space and time as Dirichlet conditions or
+Neumann conditions, in addition to other 
+convenient conditions.
+"""
+abstract type AbstractBC{FT <: AbstractFloat} end
+
+"""
+    AbstractBoundary{}
+An abstract type to indicate which boundary we are doing calculations for.
+Currently, we support the top boundary (TopBoundary)
+and bottom boundary (BottomBoundary).
+"""
+abstract type AbstractBoundary{} end
+
+
+"""
+    TopBoundary{} <: AbstractBoundary{}
+A simple object which should be passed into a function to
+indicate that we are considering the top boundary of the soil.
+"""
+struct TopBoundary <: AbstractBoundary end
+
+"""
+    BottomBoundary{} <: AbstractBoundary{}
+A simple object which should be passed into a function to
+indicate that we are considering the bottom boundary of the soil.
+"""
+struct BottomBoundary <: AbstractBoundary end
+
+"""
+    get_Δz(z)
+A function to return a tuple containing the distance between the top boundary
+and its closest center, and the bottom boundary and its closest center, 
+both as Fields.
+"""
+function get_Δz(z)
```

Comment:
Looking at this again, it might be helpful for future users if we include the type of `z` in the docstring (or specify that it should be the object returned by `Fields.coordinate_fields.z`).